module1/homework/lengthSort/sort.cpp

```diff
@@ -0,0 +1,17 @@
+#include "sort.hpp"
+
+#include <algorithm>
+
+std::deque<std::string> lengthSort(std::forward_list<std::string> list) {
+    std::deque<std::string> deque;
+    std::copy(list.begin(), list.end(), std::back_inserter(deque));
+
+    std::sort(deque.begin(), deque.end(), [](const auto& lhs, const auto& rhs) {
+        if (lhs.size() < rhs.size() || lhs < rhs) {
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
```suggestion
return lhs.size() < lhs.size() || ( lhs.size() == lhs.size() && lhs < rhs );
```

Reply:
My "if" condition is not good at all, but it's from stl homework #1. I made branch for homework #2 before I implemented the right solution here.